python/langsmith/evaluation/_runner.py

```diff
@@ -0,0 +1,895 @@
+                    # TODO: Better error
+                    f"Experiment {self.experiment_name} already exists."
+                    " Please use a different name."
+                )
+        else:
+            project = self._experiment
+        if project.url:
+            # HACKHACK
+            project_url = project.url.split("?")[0]
+            dataset_id = first_example.dataset_id
+            base_url = project_url.split("/projects/p/")[0]
+            comparison_url = (
+                f"{base_url}/datasets/{dataset_id}/compare?"
+                f"selectedSessions={project.id}"
+            )
+            print(  # noqa: T201
+                f"View the evaluation results for experiment: '{self.experiment_name}'"
+                f" at:\n{comparison_url}\n\n"
+            )
+        else:
+            # HACKHACK
+            print("Starting evaluation of experiment: %s", self.experiment_name)
+        return _ExperimentManager(
+            _examples,
+            experiment=project,
+            metadata=self._metadata,
+            client=self.client,
+            runs=self._runs,
+            evaluation_results=self._evaluation_results,
+        )
+
+    def with_predictions(
+        self,
+        target: TARGET_T,
+        /,
+        max_concurrency: Optional[int] = None,
+    ) -> _ExperimentManager:
+        context = copy_context()
+        _experiment_results = context.run(
+            self._predict, target, max_concurrency=max_concurrency
+        )
+        r1, r2 = itertools.tee(_experiment_results, 2)
+        return _ExperimentManager(
+            (pred["example"] for pred in r1),
+            experiment=self._experiment,
+            metadata=self._metadata,
+            client=self.client,
+            runs=(pred["run"] for pred in r2),
+            # TODO: Can't do multiple prediction rounds rn.
+            # evaluation_results=(
+            #     pred["evaluation_results"] for pred in _experiment_results
+            # ),
+        )
+
+    def with_scores(
+        self,
+        evaluators: Sequence[
+            Union[
+                EVALUATOR_T,
+                RunEvaluator,
+            ]
+        ],
+        *,
+        max_concurrency: Optional[int] = None,
+    ) -> _ExperimentManager:
+        evaluators = _resolve_evaluators(evaluators)
+        context = copy_context()
+        experiment_results = context.run(
+            self._score, evaluators, max_concurrency=max_concurrency
+        )
+        r1, r2, r3 = itertools.tee(experiment_results, 3)
+        return _ExperimentManager(
+            (result["example"] for result in r1),
+            experiment=self._experiment,
+            metadata=self._metadata,
+            client=self.client,
+            runs=(result["run"] for result in r2),
+            evaluation_results=(result["evaluation_results"] for result in r3),
+            aggregate_results=self._aggregate_results,
+        )
+
+    def with_batch_scores(
```

Comment:
We should not use batch -- confusing terminology. Know this is internal api but still. Let's consolidate between batch / summary